freelist.go

```diff
@@ -349,6 +349,28 @@ func (f *freelist) reload(p *page) {
 	f.readIDs(a)
 }
 
+// noSyncReload reads the freelist from pgids and filters out pending items.
+func (f *freelist) noSyncReload(pgids []pgid) {
```

Comment:
freelist knows its own type. we can do the branch inside the reload func.

Reply:
The type of freelist is hashmap or array. It is irrelevant with sync or not-sync.